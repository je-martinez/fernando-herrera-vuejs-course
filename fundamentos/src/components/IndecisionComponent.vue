<template>
  <img v-if="image" :src="image" alt="Vue logo" />
  <div class="bg-dark"></div>
  <div class="indecision-container">
    <input type="text" placeholder="Hazme una pregunta" v-model="question" />
    <p>Recuerda terminar con un signo de interrogacion (?)</p>
    <div v-if="isValidQuestion">
      <h2>{{ question }}</h2>
      <h1>{{ answer }}</h1>
    </div>
  </div>
</template>

<script>
export default {
  data() {
    return {
      question: "",
      answer: "",
      image: undefined,
      isValidQuestion: false,
    };
  },
  methods: {
    async getAnswer() {
      this.answer = "Pensando...";
      const { answer, image } = await fetch("https://yesno.wtf/api").then(
        (res) => res.json()
      );
      const messages = {
        yes: "SÃ­",
        no: "No",
        maybe: "Tal vez",
      };
      this.answer = messages[answer];
      this.image = image;
    },
  },
  watch: {
    question(value) {
      this.isValidQuestion = false;
      if (!value?.includes("?")) return;
      this.isValidQuestion = true;
      this.getAnswer();
    },
  },
};
</script>

<style scoped></style>
