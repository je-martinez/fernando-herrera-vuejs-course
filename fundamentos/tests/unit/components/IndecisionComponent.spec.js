import IndecisionComponent from "@/components/IndecisionComponent.vue";
import { shallowMount } from "@vue/test-utils";

describe("IndecisionComponent.vue", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(IndecisionComponent);
  });

  it("should match with snapshot default value", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
