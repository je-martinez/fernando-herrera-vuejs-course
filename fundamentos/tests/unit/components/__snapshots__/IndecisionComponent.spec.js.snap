// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndecisionComponent.vue should match with snapshot default value 1`] = `
<!--v-if-->
<div class="bg-dark"></div>
<div class="indecision-container"><input type="text" placeholder="Hazme una pregunta">
  <p>Recuerda terminar con un signo de interrogacion (?)</p>
  <!--v-if-->
</div>
`;
